### DISCUSSION TITLE
Reshape also change the value?

### DISCUSSION THREAD

**Discussion by UrDailyCs** [mrdbourke/pytorch-deep-learning#303] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/65452176/218776181-45753fcd-bfba-4eca-9b22-30201cd8cbe2.png)

so from your vidoe explanation you says that view changes the tensor value and thats what differentiate `view `from `reshape.` but,as you can see, not only `view `changes the tensor value but reshape also does change the tensor value. based on this [link](https://stackoverflow.com/questions/49643225/whats-the-difference-between-reshape-and-view-in-pytorch). it says that `reshape `also shares memory with the original tensor. so whats the different between the 2?

**Comment by nazarPuriy** [comment, marked as answer]
In general purposes there are the same. If want independent data add .clone() or multiply by .... = reshape(...)*1
https://stackoverflow.com/questions/49643225/whats-the-difference-between-reshape-and-view-in-pytorch

**Comment by mrdbourke** [comment]
Hi @UrDailyCs,

@nazarPuriy is correct.

For most purposes they are the same.

However, this is one of the main differences (shared memory):

> If you just want to reshape tensors, use `torch.reshape`. If you're also concerned about memory usage and want to ensure that the two tensors share the same data, use `torch.view`.

See this answer for more details: https://stackoverflow.com/a/54507446/7900723
